app/assets/js/app/modules/totalliser.js

```diff
@@ -0,0 +1,71 @@
+import domready from './domready'
+import {forEach} from 'lodash'
+import getTransitionEndEvent from '../helpers/transitionend'
+
+const transitionendEvent = getTransitionEndEvent()
+
+domready(() => {
+  const breakdownFields = document.querySelectorAll('.answer:not(.calculated) input')
+  const totalField = document.querySelectorAll('.calculated input')[0]
+
+  const onAnswerChanged = () => {
+    let newTotal = 0
+    let count = breakdownFields.length
+
+    let i
+    for (i = 0; i < count; i++) {
+      let percentage = breakdownFields[i].value
+
+      let value = parseInt(percentage)
+      if (isNaN(value)) {
+        continue
+      } else {
+        newTotal += value
+      }
+    }
+
+    updateTotal(newTotal)
+  }
+
+  const highlight = () => {
+    if (transitionendEvent) {
+      totalField.addEventListener(transitionendEvent, onTransitionHighlightEnd, true)
+    } else {
+      window.setTimeout(onTransitionHighlightEnd, 500)
+    }
+    totalField.classList.add('highlighted')
+  }
+
+  const removeHighlight = () => {
+    if (transitionendEvent) {
+      totalField.addEventListener(transitionendEvent, onTransitionHighlightEnd, true)
+    } else {
+      window.setTimeout(onTransitionHighlightEnd, 500)
+    }
+    totalField.classList.remove('highlighted')
+  }
+
+  const onTransitionHighlightEnd = (e) => {
+    totalField.removeEventListener(transitionendEvent, onTransitionHighlightEnd)
+  }
+
+  const updateTotal = (value) => {
+    totalField.value = value
+    if (value > 100 || value < 0) {
+      // bad
+      highlight()
+    } else if (value === 100) {
+      // good
+      removeHighlight()
+    } else if (value < 100 && value >= 0) {
+      // normal
+      highlight()
+    }
+  }
+
+  forEach(breakdownFields, (value) => {
+    value.addEventListener('change', onAnswerChanged)
```

Comment:
@hamishtaplin I may be totally wrong here, but my understanding after reading that article is that we could add a single event listener to a surrounding element, but is that really what we want? I'm only interested in when any of the breakdown input fields change. Since the total field is also an Answer, if the change listener was applied to a surrounding element, it would also apply to the Total field, which is not what we want. Are the performance gains significant enough to introduce a new element (e.g. a DIV) around the breakdown fields?@hamishtaplin I may be totally wrong here, but my understanding after reading that article is that we could add a single event listener to a surrounding element, but is that really what we want? I'm only interested in when any of the breakdown input fields change. Since the total field is also an Answer, if the change listener was applied to a surrounding element, it would also apply to the Total field, which is not what we want. Are the performance gains significant enough to introduce a new element (e.g. a DIV) around the breakdown fields?